packages/dds/tree/src/simple-tree/toStoredSchema.ts

```diff
@@ -55,42 +109,39 @@ export function toStoredSchema(root: ImplicitFieldSchema): TreeStoredSchema {
 				}
 				nodeSchema.set(
 					brand(schema.identifier),
-					getStoredSchema(schema as SimpleNodeSchemaBase<NodeKind> as SimpleNodeSchema),
+					getStoredSchema(
+						schema as SimpleNodeSchemaBase<NodeKind> as SimpleNodeSchema,
+						options,
+					),
 				);
 			},
+			allowedTypeFilter: (allowedType) => allowedTypeFilter(allowedType, options),
 		});
 
 		const result: TreeStoredSchema = {
 			nodeSchema,
-			rootFieldSchema: convertField(normalized),
+			rootFieldSchema: convertField(normalized, options),
 		};
 		return result;
 	});
 }
 
-/**
- * Converts a {@link SimpleTreeSchema} into a {@link TreeStoredSchema}.
- */
-export function simpleToStoredSchema(root: SimpleTreeSchema): TreeStoredSchema {
```

Comment:
extractPersistedSchema used to use this, but it caused issue due to it not being clear which node schema should be included. Including all from the provided tree schema made the most sense, however that does not make sense with staged schema filters. At least for now, to ensure consistency extractPersistedSchema is using toStoredSchema above,